### PR TITLE
SourceIRC support

### PR DESCRIPTION
**Pull request by vintagepc**
Redone version of #91 but without the need for `#tryinclude` and with a bonus cvar

Again, feel free to trashbin or do whatever you like with the other PR :smile:

### CODE REVIEW

File: scripting/lilac/lilac_stock.sp [hunk 1]
```diff
@@ -132,6 +132,19 @@ void lilac_log(bool cleanup)
 	}
 
 	WriteFileLine(file, "%s", line);
+	// Just echo log lines to SourceIRC
+	if (icvar[CVAR_SOURCEIRC] && NATIVE_EXISTS("IRC_MsgFlaggedChannels")) {
+			// Note- SourceIRC Expects messages to be clean with no \r or \n, so clean it if not already done.
```
Comment (J-Tanzanite) [inline scripting/lilac/lilac_stock.sp:137]: I can't help but notice the double indentation here?
Might need a slight correction ;)

**Review by J-Tanzanite — CHANGES_REQUESTED** [review]
Other than this oddity, the PR seems great!
I was able to compile without SourceIRC (obviously), and all runs well :)

Since I've never used SourceIRC, I haven't tested if this works, but I don't see why it wouldn't.

File: scripting/lilac/lilac_stock.sp (hunk 1 above)
Comment (vintagepc) [reply scripting/lilac/lilac_stock.sp]: Fixed, not sure what happened there, either bad editor indent detection or bad copy-paste :)

**Comment by J-Tanzanite** [comment]
Thanks, all looks good now! Merged :)

**Pull request merged**